### PR TITLE
Full switch to poetry

### LINKED ISSUE #289: Fully migrate to `poetry`
This should more permanently deal with problems such as #288.

Super low priority.

### LINKED ISSUE #467: Get rid of dependency on requests
The requests library is thusly sparsely used in this project that it's probably nicer to just use the Python standard library here.

### LINKED ISSUE #587: 'ABCMeta' object is not subscriptable on Python 3.8.10
On Python 3.8.10 I get:
`TypeError: 'ABCMeta' object is not subscriptable`

I have fixed it by replacing:
`if sys.version_info >= (3, 8) or TYPE_CHECKING:`
with:
`if sys.version_info >= (3, 9) or TYPE_CHECKING:`
in all places,
but I'm not sure what is the correct minimum version.

### PR DESCRIPTION
**Pull request by carmenbianca**
Accidentally closed #599 due to wrong branch name.

Fixes #467, fixes #289, fixes #587

I was doing some package maintenance for Fedora. This should make my life a whole lot easier.

~~Maybe postpone this for 2.0 major release.~~ No longer a breaking change. Python 3.6 remains supported.

TODO:

- [x] Translation build step.
- [x] Don't install dev requirements in Docker images and reuse GitHub workflow
- [x] Fix Docker containers
- [x] Fix documentation
- [x] Change log
- [ ] Write message to reuse mailing list
- [x] Test poetry publishing
  - [ ] Signed releases?
- [x] Fix python-debian dependency
  - [ ] Ideally, ship with dep5 parser microlibrary.
- [x] Replicate and document `make test` / `make lint`.
- [x] Fix pylint pre-commit.
- [x] Remove requests dependency now that we're messing with dependencies anyway.

### CODE REVIEW

**Comment by carmenbianca** [comment]
This is effectively ready for review. It all works without any breaking changes. The only thing I haven't tested is `poetry publish`.

**Comment by carmenbianca** [comment]
`poetry publish` tested. It works, with two caveats:

- Signing with GPG keys not possible.
- Because of our `build.py` step, the resulting wheel is not 'noarch'. This is incorrect, but I've not found a way to fix it. There probably will be a way to fix this once Poetry starts stabilising and documenting the build plugin system.

File: README.md [hunk 1]
```diff
@@ -262,14 +262,16 @@ Starting local development is very simple, just execute the following commands:
 ```bash
 git clone git@github.com:fsfe/reuse-tool.git
 cd reuse-tool/
-python3 -mvenv venv
-source venv/bin/activate
-make develop
+poetry install  # You may need to install poetry using your package manager.
+poetry run pre-commit install  # Using poetry is optional here if you already have pre-commit.
```
Comment (carmenbianca) [inline README.md:266]: Add `pre-commit install`

**Comment by nicorikken** [comment]
I just tested the `poetry install` and `poetry run reuse ...` commands and it works well. For me this simplifies setting up a development environment. Thanks a lot!

**Comment by carmenbianca** [comment]
Failing test because of https://github.com/pytest-dev/pytest-cov/issues/564

**Comment by carmenbianca** [comment]
I'm going to just merge this. Rebasing this branch frequently is rather time-consuming, and I've tested this thoroughly enough to know that it just works.

**Comment by linozen** [comment]
@carmenbianca Thanks! Sorry for not getting on this earlier.

**Comment by carmenbianca** [comment]
@linozen No worries! For the longest time, I maintained this project solo. Having more input is truly amazing, but by no means a blocker to keep the ball slowly moving :) Just let me know if I do something truly stupid.

**Comment by grawlinson** [comment]
This seems to break `python -m build` on Arch Linux.

**Comment by carmenbianca** [comment]
@grawlinson 1.1.0 was released yesterday. It changed the build system.

**Comment by grawlinson** [comment]
The `build` module is a PEP-517 compatible build backend that invokes the specified PEP-517 build method, and is what we use on Arch instead of the various PEP-517 tools (flit, poetry, etc).

Shall I open an issue?

**Comment by carmenbianca** [comment]
I see. Strange. Please open an issue.

**Pull request merged**